frontend/src/components/CreateEditSongbook.tsx

```diff
@@ -0,0 +1,172 @@
+import {
+  Box,
+  Button,
+  Flex,
+  FormLabel,
+  Input,
+  Modal,
+  ModalBody,
+  ModalCloseButton,
+  ModalContent,
+  ModalFooter,
+  ModalHeader,
+  ModalOverlay,
+} from "@chakra-ui/react";
+import { AxiosResponse } from "axios";
+import { useEffect, useState } from "react";
+import { UseAsyncReturn } from "react-async-hook";
+import { useNavigate } from "react-router-dom";
+import { Songbook } from "../models";
+import { createNewSongbook, editSongbook } from "../services/songs";
+
+interface CreateEditSongbookProps {
+  isOpen: boolean;
+  is_noodle_mode: boolean | undefined;
+  onClose: () => void;
+  asyncSongbook: UseAsyncReturn<AxiosResponse<Songbook, any>, never[]> | null;
+}
+
+export default function CreateEditSongbook({
+  isOpen,
+  onClose,
+  asyncSongbook,
+  is_noodle_mode,
+}: CreateEditSongbookProps) {
+  const [maxSongs, setMaxSongs] = useState<string>("");
```

Comment:
I think we want this to be typed as `number?` and defaulted to null, then in the onChange callback we should check if the new val is null and if not we should ensure that the value is a parse-able int before storing it (as a number) in maxSongs.

Currently you can enter non-numbers into the field, which is what I'd like to avoid with the above comment.